sample/rules.go

```diff
@@ -159,15 +162,21 @@ func ruleMatchesTrace(t *types.Trace, rule *config.RulesBasedSamplerRule, checkN
 				matched++
 				continue
 			} else {
-				// if HasRootSpan is one of the conditions and it didn't match,
-				// there's no need to check the rest of the conditions.
 				return false
 			}
+
 		}
+		var checkedRootPrefix bool
 
 	span:
 		for _, span := range t.GetSpans() {
-			value, exists := extractValueFromSpan(t, span, condition, checkNestedFields)
+			var value interface{}
```

Comment:
```suggestion
			var value any
```
We should prefer using `any` to `interface{}` in new code.

Reply:
Will do.